vochain/transaction.go

```diff
@@ -174,8 +181,54 @@ func VoteTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State,
 	var vote *models.Vote
 	switch {
 	case process.EnvelopeType.Anonymous:
-		// TODO check snark
-		return nil, fmt.Errorf("snark vote not implemented")
+		// Supports Groth16 proof generated from circom snark compatible
+		// prover
+		proofZkSNARK := ve.Proof.GetZkSnark()
+		if proofZkSNARK == nil {
+			return nil, fmt.Errorf("zkSNARK proof is empty")
+		}
+		proof, publicInputsFromUser, err := zk.ProtobufZKProofToCircomProof(proofZkSNARK)
+		if err != nil {
+			return nil, err
+		}
+
+		verificationKey := app.ZkVks[proofZkSNARK.CircuitParametersIndex]
+		if verificationKey == nil {
+			return nil, fmt.Errorf("VerificationKey==nil")
+		}
```

Comment:
This should never happen because app.ZkVks is  slice, and we should never add a nil pointer to that slice.